src/Xamarin.Android.Build.Tasks/Microsoft.Android.Sdk/targets/Microsoft.Android.Sdk.NativeRuntime.targets

```diff
@@ -0,0 +1,69 @@
+<!--
+***********************************************************************************************
+Microsoft.Android.Sdk.NativeRuntime.targets
+
+Contains code to build and link the native runtime at application build time.
+
+***********************************************************************************************
+-->
+<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <UsingTask TaskName="Xamarin.Android.Tasks.GetNativeRuntimeComponents" AssemblyFile="$(_XamarinAndroidBuildTasksAssembly)" />
+  <UsingTask TaskName="Xamarin.Android.Tasks.LinkNativeRuntime" AssemblyFile="$(_XamarinAndroidBuildTasksAssembly)" />
+
+  <Target Name="_PrepareUnifiedNativeRuntimeItems"
+          Condition=" '$(_AndroidEnableNativeRuntimeLinking)' == 'true' ">
+    <ItemGroup>
+      <_UnifiedNativeRuntime Include="$(_AndroidApplicationSharedLibraryPath)%(_BuildTargetAbis.Identity)\libnetdroid-unified.so">
+        <abi>%(_BuildTargetAbis.Identity)</abi>
+        <ArchiveFileName>libmonodroid.so</ArchiveFileName>
+      </_UnifiedNativeRuntime>
+    </ItemGroup>
+
+    <ItemGroup>
+      <_ResolvedNativeArchive Include="@(ResolvedFileToPublish)" Condition=" '%(ResolvedFileToPublish.Extension)' == '.a' " />
+      <_ResolvedNativeObjectFile Include="@(ResolvedFileToPublish)" Condition=" '%(ResolvedFileToPublish.Extension)' == '.o' " />
+      <_ApplicationSharedLibrary Include="@(_UnifiedNativeRuntime)" />
+    </ItemGroup>
+
+    <GetNativeRuntimeComponents
+        MonoComponents="@(_MonoComponent)"
+        ResolvedNativeObjectFiles="@(_ResolvedNativeObjectFile)"
+        ResolvedNativeArchives="@(_ResolvedNativeArchive)"
+        HackLocalClrRepoPath="$(_HackLocalClrRuntimePath)">
+      <Output TaskParameter="NativeArchives" ItemName="_SelectedNativeArchive" />
+      <Output TaskParameter="RequiredLibraries" ItemName="_RequiredLinkLibraries" />
+      <Output TaskParameter="LinkStartFiles" ItemName="_NativeLinkStartFiles" />
+      <Output TaskParameter="LinkEndFiles" ItemName="_NativeLinkEndFiles" />
+      <Output TaskParameter="NativeSymbolsToExport" ItemName="_NativeSymbolsToExport" />
+    </GetNativeRuntimeComponents>
+  </Target>
+
+  <Target Name="_LinkNativeRuntime"
+          DependsOnTargets="_CompileNativeAssemblySources;_PrepareUnifiedNativeRuntimeItems"
+          Inputs="@(_NativeAssemblyTarget);@(_SelectedNativeArchive)"
+          Outputs="@(_UnifiedNativeRuntime)"
+          Condition=" '$(_AndroidEnableNativeRuntimeLinking)' == 'true' ">
+
+    <!-- AndroidApiLevel must be set to the same version our runtime, MonoVM and CoreCLR are built
+         against -->
+    <LinkNativeRuntime
+        MonoComponents="@(_MonoComponent)"
+        AndroidBinUtilsDirectory="$(AndroidBinUtilsDirectory)"
+        AndroidNdkDirectory="$(_AndroidNdkDirectory)"
+        AndroidApiLevel="$(AndroidMinimumSupportedApiLevel)"
```

Comment:
Should the property be `AndroidMinimumApiLevel`, so it would be clear in the C# task that it's 21 or some lower number?

Reply:
Do you mean `AndroidMinimumDotNetApiLevel`? It's not available at application build time I think, but even if it was, we couldn't use it. The `AndroidApiLevel` refers to what was detected in the NDK existing on the user's machine - our `AndroidMinimumDotNetApiLevel` might not be found in that copy of NDK and the link would fail.

However, I've just realized that we might not need this property at all, since we now redistribute the NDK libraries and object files required for linking. I'll work on this today.